tests/time_tests/run_executable.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+This script runs TimeTests executable several times to aggregate
+collected statistics.
+"""
+
+import statistics as Statistics
+from pathlib import Path
+import tempfile
+import subprocess
+import logging
+import argparse
+import sys
+import yaml
+
+
+def run_cmd(args: list, log=None, verbose=True):
+    """ Run command
+    """
+    if log is None:
+        log = logging.getLogger('run_cmd')
+    log_out = log.info if verbose else log.debug
+
+    log.info(f'========== cmd: {" ".join(args)}')  # pylint: disable=logging-format-interpolation
+
+    proc = subprocess.Popen(args,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                            encoding='utf-8',
+                            universal_newlines=True)
+    output = []
+    for line in iter(proc.stdout.readline, ''):
+        log_out(line.strip('\n'))
+        output.append(line)
+        if line or proc.poll() is None:
+            continue
+        break
+    outs = proc.communicate()[0]
+
+    if outs:
+        log_out(outs.strip('\n'))
+        output.append(outs)
+    log.info('========== Completed. Exit code: %d', proc.returncode)
+    return proc.returncode, ''.join(output)
+
+
+def read_statistics(statistics_path, statistics: dir):
+    """Read statistics from a file and extend provided statistics"""
+    with open(statistics_path, "r") as file:
+        parsed_data = yaml.load(file, Loader=yaml.FullLoader)
+    return dict((step_name, statistics.get(step_name, []) + [duration])
+                for step_name, duration in parsed_data.items())
+
+
+def aggregate_statistics(statistics: dir):
+    """Aggregate provided statistics"""
+    return {step_name: {"avg": Statistics.mean(duration_list),
+                        "stdev": Statistics.stdev(duration_list)}
+            for step_name, duration_list in statistics.items()}
+
+
+def write_aggregated_stats(statistics_path, statistics: dir):
+    """Write aggregated statistics to a file in YAML format"""
+    with open(statistics_path, "w") as file:
+        yaml.dump(statistics, file)
+
+
+def run_executable(executable: Path, model: Path, device, niter, log=None):
+    """Run provided executable several times and aggregate collected statistics"""
+
+    if log is None:
+        log = logging.getLogger('run_executable')
+
+    cmd_common = [str(executable.resolve()),
+                  "-m", str(model.resolve()),
+                  "-d", device]
+
+    # Create folder to save statistics files
+    statistics_dir = (Path(".") / "statistics_dir").absolute()
+    statistics_dir.mkdir(parents=True, exist_ok=True)
+
+    # Run executable and collect statistics
+    statistics = {}
+    for run_iter in range(niter):
+        statistics_path = statistics_dir / Path(tempfile.NamedTemporaryFile().name).stem
```

Comment:
Consider using mkstemp instead. The NamedTemporaryFile is a wrapper to support auto-deletion which you are not using here.

Reply:
I see no reason to use `mkstemp` instead. I just use a random-generated filename provided by `tempfile` module to save files in `statistics_dir` local folder. In this case auto-deletion is required in order to not to save temp files

If you have any ideas how to avoid auto-generation of temp files, it will be good to implement or use

Reply:
Yes, it's really what I wanted to use, thanks!

Reply:
Done